dhcpv4/client.go

```diff
@@ -118,18 +162,30 @@ func MakeListeningSocket(ifname string) (int, error) {
 	return fd, nil
 }
 
-// Exchange runs a full DORA transaction: Discover, Offer, Request, Acknowledge,
+// Exchange is a shortcut for ExchangeWithOptions when you want to use the default
+// option set.
+func (c *Client) Exchange(ifname string, discover *DHCPv4, modifiers ...Modifier) ([]*DHCPv4, error) {
+	return c.ExchangeWithOptions(ifname, discover, &ClientExchangeOptions{}, modifiers...)
+}
+
+
+// ExchangeWithOptions runs a full DORA transaction: Discover, Offer, Request, Acknowledge,
 // over UDP. Does not retry in case of failures. Returns a list of DHCPv4
 // structures representing the exchange. It can contain up to four elements,
 // ordered as Discovery, Offer, Request and Acknowledge. In case of errors, an
 // error is returned, and the list of DHCPv4 objects will be shorted than 4,
 // containing all the sent and received DHCPv4 messages.
-func (c *Client) Exchange(ifname string, discover *DHCPv4, modifiers ...Modifier) ([]*DHCPv4, error) {
+func (c *Client) ExchangeWithOptions(ifname string, discover *DHCPv4, options *ClientExchangeOptions, modifiers ...Modifier) ([]*DHCPv4, error) {
```

Comment:
I would rename it back to Exchange.

Reply:
I split this in another function to keep backward compatibility (so the signature of Exchange() does not change). I have no problem to actually add the parameter directly to the function it if we don't care much.

Reply:
Do we actually need the parameter? I think we can check if we need to use broadcast or unicast by checking if the RemoteAddr is set.

Reply:
It depends how much freedom we want to give library users. I have no hard feelings either way.

Reply:
fair point - I replied already on the other comment thread, basically I'd say: let's start simple as Pablo suggests (and ignore my initial request to create a struct), then we can expand it in a separate PR, to keep things simple. Would that make sense?

Reply:
It does, I'll update the code 👍